### PR TITLE
:art:  Add color support to IDisplayable interface

### LINKED ISSUE #27: :art: Update IDisplayable with getter and setter for colors
# 👮‍♂️ What's needed
We need to add a getter and a setter for all the IDisplayable and not only the primitives.

# 📜 Sources
No source file.

### PR DESCRIPTION
**Pull request by Marius-P1**
I moved the getters and setters method for the color from [IPrimitive](https://github.com/epitech-mirroring/ArcadeShared/blob/main/displayable/primitives/IPrimitive.hpp) to [IDisplayable](https://github.com/epitech-mirroring/ArcadeShared/blob/main/displayable/IDisplayable.hpp)

### CODE REVIEW

**Review by axel-eck — APPROVED** [review]

File: displayable/IDisplayable.hpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 #pragma once
 #include "../utils/ICoordinate.hpp"
+#include "../utils/IColor.hpp"
```
Comment (aimericdrk) [inline displayable/IDisplayable.hpp:11]: you stinky

File: displayable/IDisplayable.hpp [hunk 2]
```diff
@@ -19,4 +20,7 @@ class IDisplayable {
     virtual void setPosition(std::unique_ptr<ICoordinate> position) = 0;
     virtual void setSize(float size) = 0;
     [[nodiscard]] virtual char getReplacingChar() const = 0;
+    [[nodiscard]] virtual const IColor &getColor() const = 0;
```
Comment (aimericdrk) [inline displayable/IDisplayable.hpp:23]: --

File: displayable/IDisplayable.hpp [hunk 3]
```diff
@@ -19,4 +20,7 @@ class IDisplayable {
...
+    virtual void setColor(const IColor &color) = 0;
```
Comment (aimericdrk) [inline displayable/IDisplayable.hpp:24]: dirty

File: displayable/IDisplayable.hpp [hunk 4]
```diff
@@ -19,4 +20,7 @@ class IDisplayable {
...
+    virtual void setColor(std::unique_ptr<IColor> color) = 0;
```
Comment (aimericdrk) [inline displayable/IDisplayable.hpp:25]: why ?

File: displayable/primitives/IPrimitive.hpp [hunk 5]
```diff
@@ -9,12 +9,8 @@
 #pragma once
 #include <memory>
 #include "../IDisplayable.hpp"
-#include "../../utils/IColor.hpp"
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp:12]: ugly

File: displayable/primitives/IPrimitive.hpp [hunk 6]
```diff
@@ -9,12 +9,8 @@
...
 
 class IPrimitive: public virtual IDisplayable {
 public:
-    ~IPrimitive() override = default;
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp:16]: pls remove your third chromosome

File: displayable/primitives/IPrimitive.hpp [hunk 7]
```diff
@@ -9,12 +9,8 @@
...
-    [[nodiscard]] virtual const IColor &getColor() const = 0;
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp:17]: useless like you

**Review by Antoinegtir — APPROVED** [review]

File: displayable/primitives/IPrimitive.hpp [hunk 8]
```diff
@@ -9,12 +9,8 @@
...
-    virtual void setColor(const IColor &color) = 0;
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp:18]: why ?

File: displayable/primitives/IPrimitive.hpp [hunk 9]
```diff
@@ -9,12 +9,8 @@
...
-    virtual void setColor(std::unique_ptr<IColor> color) = 0;
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp:19]: i was using that you bad person

File: displayable/primitives/IPrimitive.hpp [hunk 10]
```diff
@@ -9,12 +9,8 @@
...
+    ~IPrimitive() override = default;;
```
Comment (aimericdrk) [inline displayable/primitives/IPrimitive.hpp]: no

**Review by aimericdrk — APPROVED** [review]

**Pull request merged**